src/components/annual-report/report-data/section4/useFetchSectionFourData.js

```diff
@@ -0,0 +1,111 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useState } from 'react'
+
+const section4ByLevel = {
+    data_detail_by_level: {
```

Comment:
It would be great to use the constants you have set up in `section4Calculations.js` like `const OVERALL_RESPONSE_NAME = 'data_over_all_org_units'` as the keys here -- maybe define and export them in this file and import them in `section4Calculations.js`